src/katsdpcontroller/product_config.py

```diff
@@ -1396,6 +1397,81 @@ def from_config(
         )
 
 
+class GpucbfTiedArrayResampledVoltageStream:
+    """Real tied-array-resampled-voltage stream (GPU V-engine)."""
+
+    stream_type: ClassVar[str] = "gpucbf.tied_array_resampled_voltage"
+    _valid_src_types: ClassVar[_ValidTypes] = {"gpucbf.tied_array_channelised_voltage"}
+
+    def __init__(
+        self,
+        name: str,
+        src_streams: Sequence[Stream],
+        *,
+        n_chans: int,
+        pols: Sequence[str],
+        station_id: str,
+    ):
+        if len(src_streams) != 2:
+            raise ValueError(f"{self.stream_type} requires two {self._valid_src_types}")
+        grandparent_streams: List[GpucbfAntennaChannelisedVoltageStream] = [
+            tacv_src_streams.src_streams[0]
+            for tacv_src_streams in src_streams
+            if len(tacv_src_streams.src_streams) == 1
+            and isinstance(tacv_src_streams.src_streams[0], GpucbfAntennaChannelisedVoltageStream)
+        ]
+        _grandparent_stream_type = "gpucbf.antenna_channelised_voltage"
+        if len(grandparent_streams) != 2:
+            # Should be two in total, one per src_stream
+            raise ValueError(
+                f"{self.stream_type} requires {self._valid_src_types} with a single src_stream"
+            )
+        if grandparent_streams[0] != grandparent_streams[1]:
+            raise ValueError(
+                f"{self._valid_src_types} src_streams need to have the same parent "
+                f"{_grandparent_stream_type} stream"
+            )
+
+        try:
+            getattr(getattr(grandparent_streams[0], "narrowband"), "vlbi")
+        except AttributeError:
+            raise ValueError(
+                f"Grandparent {_grandparent_stream_type} stream is not configured for VLBI"
+            ) from None
+
+        output_bandwidth = grandparent_streams[0].bandwidth
+        # We have confirmed above that the vlbi attribute is not None
+        pass_bandwidth = grandparent_streams[0].narrowband.vlbi.pass_bandwidth  # type: ignore
+        actual_bandwidth_ratio = Fraction(output_bandwidth) / Fraction(pass_bandwidth)
+        expected_bandwidth_ratio = Fraction(107, 64)
+        if actual_bandwidth_ratio != expected_bandwidth_ratio:
+            raise ValueError(
+                f"Output to pass bandwidth ratio {actual_bandwidth_ratio}, "
+                f"expected {expected_bandwidth_ratio}"
+            )
+
+        self.n_chans = n_chans
```

Comment:
You're missing the call to `super().__init__(...)`.

Reply:
Definitely don't inherit from CbfStream: that's for *input* streams the SDP is receiving from outside the product controller.

I don't think CbfPerChannelStream is a great fit either - it's desired for streams where the bandwidth is split over multicast endpoints. The current plan is that the tarv stream will have only 2 channels, and they'll just go to the same multicast endpoint (or possibly several endpoints, but just for spine load-balancing rather than for substreaming).